modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonClientCodegen.java

```diff
@@ -402,18 +466,62 @@ public String modelTestFileFolder() {
         return outputFolder + File.separatorChar + testFolder;
     }
 
-    @Override
-    public String getTypeDeclaration(Schema p) {
-        if (ModelUtils.isArraySchema(p)) {
+    /**
+     * Output the type declaration of the property
+     *
+     * @param schema property schema
+     * @return a string presentation of the property type
+     */
+    public String getSimpleTypeDeclaration(Schema schema) {
+        String oasType = getSchemaType(schema);
+        if (typeMapping.containsKey(oasType)) {
+            return typeMapping.get(oasType);
+        }
+        return oasType;
+    }
+
+    public String getTypeString(Schema p, String prefix, String suffix) {
+        // this is used to set dataType, which defines a python tuple of classes
+        String typeSuffix = suffix;
+        if (suffix == ")") {
+            typeSuffix = "," + suffix;
+        }
+        if (ModelUtils.isNullable(p)) {
+            typeSuffix = ", none_type" + suffix;
+        }
+        if (ModelUtils.isFreeFormObject(p) && ModelUtils.getAdditionalProperties(p) == null) {
+            if (!ModelUtils.isNullable(p)) {
+                typeSuffix = suffix;
+            }
+            return prefix + "bool, date, datetime, dict, float, int, list, str" + typeSuffix;
+        }
+        if (ModelUtils.isMapSchema(p)) {
+            Schema inner = ModelUtils.getAdditionalProperties(p);
+            return prefix + "{str: " + getTypeString(inner, "(", ")") + "}" + typeSuffix;
+        } else if (ModelUtils.isArraySchema(p)) {
             ArraySchema ap = (ArraySchema) p;
             Schema inner = ap.getItems();
-            return getSchemaType(p) + "[" + getTypeDeclaration(inner) + "]";
-        } else if (ModelUtils.isMapSchema(p)) {
-            Schema inner = ModelUtils.getAdditionalProperties(p);
+            return prefix + "[" + getTypeString(inner, "(", ")") + "]" + typeSuffix;
+        }
+        String baseType = getSimpleTypeDeclaration(p);
+        if (baseType == "file") {
+            baseType = "file_type";
+        }
+        return prefix + baseType + typeSuffix;
+    }
 
-            return getSchemaType(p) + "(str, " + getTypeDeclaration(inner) + ")";
+    @Override
+    public String getTypeDeclaration(Schema p) {
+        // this is used to set dataType, which defines a python tuple of classes
+        return getTypeString(p, "", "");
+    }
+
+    @Override
+    public String toInstantiationType(Schema property) {
+        if (property instanceof ArraySchema || property instanceof MapSchema || property.getAdditionalProperties() != null) {
```

Comment:
Please use ModelUtils.isArraySchema instead of `property instanceof ArraySchema`. Same for the map schema check using instanceof at the moment.

Reply:
Implemented this feedback here https://github.com/spacether/openapi-generator/tree/issue_1991_uses_class_types